src/main/java/com/github/manolo8/darkbot/gui/nearplayers/NearPlayerEditor.java

```diff
@@ -0,0 +1,100 @@
+package com.github.manolo8.darkbot.gui.nearplayers;
+
+import com.github.manolo8.darkbot.Main;
+import com.github.manolo8.darkbot.config.PlayerInfo;
+import net.miginfocom.swing.MigLayout;
+
+import javax.swing.*;
+import java.util.Arrays;
+
+import static com.github.manolo8.darkbot.Main.API;
+
+public class NearPlayerEditor extends JPanel {
+
+    private final JList<PlayerInfo> nearPlayerInfoList;
+    private final DefaultListModel<PlayerInfo> nearPlayersModel;
+    protected Main main;
+
+    public NearPlayerEditor() {
+        super(new MigLayout("ins 0, gap 0, wrap 0", "[][]"));
+
+        JPanel panel = new JPanel(new MigLayout("ins 0"));
+        JButton invite = new JButton("Invite");
+        JButton save = new JButton("Save");
+        nearPlayerInfoList = new JList<>(nearPlayersModel = new DefaultListModel<>());
+        nearPlayerInfoList.setCellRenderer(new NearPlayerRenderer());
+
+        panel.add(invite);
+        panel.add(save);
+        add(panel, "split 2, wrap, align center");
+        add(new JScrollPane(nearPlayerInfoList,
+                JScrollPane.VERTICAL_SCROLLBAR_AS_NEEDED, JScrollPane.HORIZONTAL_SCROLLBAR_NEVER), "grow");
+
+
+//        nearPlayersModel.add(0, new PlayerInfo("asd", 123));
+//        nearPlayersModel.add(1, new PlayerInfo("asdd", 1233));
+//        nearPlayersModel.add(2, new PlayerInfo("asddd", 12333));
+
+        nearPlayerInfoList.addListSelectionListener(e -> {
+            if (!e.getValueIsAdjusting())
+                if (nearPlayerInfoList.getSelectedValue() != null)
+                    System.out.println("selected: " + nearPlayerInfoList.getSelectedValue().userId + " " + nearPlayerInfoList.getSelectedValue().username);
+        });
+        invite.addActionListener(e ->
+        {
+            //TODO: add direct invite function
+            //new GroupManager(main).trySendInvite(nearPlayerInfoList.getSelectedValue().username);
+        });
+        save.addActionListener(e ->
+        {
+            //TODO: add save function to player tag list
+        });
+        refreshList();
+    }
+
+
+    public void refreshList() {
+        Thread t1 = new Thread(() -> {
+            while (true) {
+                if (main != null) {
```

Comment:
You're creating a few threading issues here
 - The entity updates must only be accessed on the main thread, or if accessed from other threads, they must use synchronized access, otherwise you'll get concurrent modification exceptions and your thread will crash.
  - You're modifying swing components from outside the swing ui thread, that can cause inconsistencies in the models and break or deadlock the UI
  
The solution is to make use of entity events (launched from the main thread) and using `SwingUtilities.invokeLater(() -> { your swing code });` for any ui-editing code (addElement, or remove, etc)